### PR TITLE
golangci-lint: use exclude-rules instead of skip-files and skip-dirs

### PR DESCRIPTION
**Pull request by mmorel-35**
# Please add a summary of your change

- use exclude-rules instead of skip-files and skip-dirs
- includes fixes of misspell, testifylint (len) and whitespace linters

# Does your change fix a particular issue?

No

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: golangci.yaml [hunk 1]
```diff
@@ -34,8 +32,6 @@ run:
   # "/" will be replaced by current OS file path separator to properly work
   # on Windows.
   skip-files:
```
Comment (mateusoliveira43) [inline golangci.yaml:34]: since this field has no items anymore, lets delete it

**Review by mateusoliveira43 — APPROVED** [review]
just one little change in golangci.yaml

At first this PR seemed strange for me, because we would be just skipping linter checks in test files in a different way. But it does already enforce some linters (like misspell and whitespace) to all test files, which is very good :smile:

**Review by mateusoliveira43 — APPROVED** [review]
LGTM

**Review by blackpiglet — APPROVED** [review]

**Review by danfengliu — APPROVED** [review]
LGTM

**Pull request merged**